### PR TITLE
Book store develop

### PR DESCRIPTION
**Pull request by StoneBlood-bit**

### CODE REVIEW

File: src/main/java/mate/academy/config/SecurityConfig.java [hunk 1]
```diff
@@ -40,4 +42,11 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .userDetailsService(userDetailsService)
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/config/SecurityConfig.java:48]: you missed `addFilterBefore` and `sessionManagement` configuration

File: src/main/java/mate/academy/controller/AuthenticationController.java [hunk 2]
```diff
@@ -4,20 +4,30 @@
...
+@Tag(name = "User registration", description = "Endpoints for registration and authentication user")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/auth")
 public class AuthenticationController {
     private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @Operation(summary = "login user", description = "user authentication")
+    @RequestMapping("/login")
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/controller/AuthenticationController.java]: Type of request missed!!! Check other endpoints

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/security/JwtAuthenticationFilter.java]: HttpHeader.AUTHORIZATION

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/security/JwtAuthenticationFilter.java]: Make it a constant field. and use the length method instead 7

File: src/main/java/mate/academy/security/JwtUtil.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    public boolean isValidToken(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .setSigningKey(secret)
+                    .build()
+                    .parseClaimsJws(token);
+            return !claimsJws.getBody().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            System.err.println("Invalid token: " + token + ", error: " + e.getMessage());
+            return false;
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/security/JwtUtil.java]: throw exception. If you want to log something better use SLF4J

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: pom.xml [hunk 6]
```diff
@@ -115,6 +115,25 @@
 			<version>2.7.0</version>
 		</dependency>
 
+		<dependency>
+			<groupId>io.jsonwebtoken</groupId>
+			<artifactId>jjwt-api</artifactId>
+			<version>0.12.6</version>
```
Comment (Elena-Bruyako) [inline pom.xml]: let's add version to properties

File: pom.xml [hunk 7]
```diff
@@ -115,6 +115,25 @@
...
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>io.jsonwebtoken</groupId>
+			<artifactId>jjwt-jackson</artifactId>
+			<version>0.12.6</version>
+			<scope>runtime</scope>
+		</dependency>
+
```
Comment (Elena-Bruyako) [inline pom.xml]: remove redundant empty lines

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/AuthenticationController.java [hunk 8]
```diff
@@ -4,20 +4,31 @@
...
+    private final AuthenticationService authenticationService;
+
+    @Operation(summary = "login user", description = "user authentication")
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
+        return authenticationService.authenticate(requestDto);
+    }
 
     @Operation(summary = "registration user", description = "registration a new user")
     @RequestMapping("/registration")
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/controller/AuthenticationController.java:34]: Type of request missed!!! Check other endpoints

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But fix my comment before merging ;)

**Pull request merged**